ops/model.py

```diff
@@ -230,6 +247,84 @@ def _get_unique(self, relation_name, relation_id=None):
             raise TooManyRelatedAppsError(relation_name, num_related, 1)
 
 
+class BindingMapping(Mapping):
+    """A map of binding names to lists of Binding instances."""
+
+    def __init__(self, binding_names, backend):
+        """
+        binding_names -- Relation endpoint names and names of extra-bindings.
+        backend -- An instance of a model backend implementation.
+        """
+        self._backend = backend
+        self._data = {binding_name: None for binding_name in binding_names}
+
+    def __contains__(self, key):
+        return key in self._data
+
+    def __len__(self):
+        return len(self._data)
+
+    def __iter__(self):
+        return iter(self._data)
+
+    def __getitem__(self, binding_name):
+        binding_list = self._data[binding_name]
+        if binding_list is None:
+            binding_list = self._data[binding_name] = []
+            # A relation id can be None which means a binding is not associated with a relation.
+            binding_list.append(Binding(binding_name, None, self._backend))
+            for relation_id in self._backend.relation_ids(binding_name):
+                binding_list.append(Binding(binding_name, relation_id, self._backend))
```

Comment:
This logic seems a bit suspect. Why is it returning all bindings for all relations established plus the one that isn't associated with any relation at all?  Imagine you now have three bindings, for different networks, and a binding for no relation at all. How do we expect people to make a good use of this result?